analyses/malawi/scripts/mwi_chirps_dry_spell_detection.R

```diff
@@ -85,144 +99,267 @@ for (i in seq_along(1:nbr_layers)) {
+
+rainfall_during_dry_spells_stats <- rainfall_during_dry_spells %>%
+                                      group_by(pcode, dry_spell_confirmation) %>%
+                                      summarise(n_days = n(),
+                                                dry_spell_rainfall = sum(total_prec))
+
+dry_spells_details <- dry_spells_list %>%
+                        left_join(rainfall_during_dry_spells_stats, by = c('pcode' = 'pcode', 'dry_spell_confirmation' = 'dry_spell_confirmation', 'dry_spell_duration' = 'n_days'))
+
+nrow(dry_spells_list) == nrow(dry_spells_details) # check that all records were kept
+
+# identify dry spells during rainy seasons
+dry_spells_during_rainy_season_list <- dry_spells_details %>%
+                                          left_join(rainy_seasons[, c('pcode', 'season_approx', 'onset_date', 'cessation_date')], by = c('pcode', 'season_approx'), all.x = T, all.y = T) %>% # add rainy onset and cessation dates
+                                          mutate(confirmation_date_during_rainy_season = ifelse(dry_spell_confirmation >= onset_date & dry_spell_confirmation <= cessation_date, 1, 0)) %>% # identifies dry spells that reached 14-d rolling sum during rainy season 
+                                          filter(confirmation_date_during_rainy_season == 1) %>% # only keep dry spells that were confirmed during rainy season even if started before onset or ended after cessation
+                                          dplyr::select(pcode, season_approx, dry_spell_first_date, dry_spell_last_date, dry_spell_duration, dry_spell_rainfall)
 
 # add region names for ease of communication
-dry_spells_list <- dry_spells_list %>% 
-                      left_join(mwi_adm2_ids, by = c('pcode'= 'ADM2_PCODE')) %>%
-                      dplyr::select(pcode, ADM2_EN, dry_spell_confirmation, dry_spell_first_date, dry_spell_last_date, duration_days)
-
-# summary stats per region ## FIX ME: no CHIRPS data for Likoma, Mwanza?
-dry_spells_summary_per_region <- dry_spells_list %>%
-                                    group_by(pcode, ADM2_EN) %>%
-                                    summarise(nbr_dry_spells = n(),
-                                              median_duration = mean(duration_days),
-                                              min_duration = min(duration_days),
-                                              max_duration = max(duration_days))
+dry_spells_during_rainy_season_list <- dry_spells_during_rainy_season_list %>% 
+                                          left_join(mwi_adm2_ids, by = c('pcode'= 'ADM2_PCODE')) %>%
+                                          dplyr::select(pcode, ADM2_EN, dry_spell_first_date, dry_spell_last_date, dry_spell_duration, dry_spell_rainfall)
+
+write.csv(dry_spells_during_rainy_season_list, file = paste0(data_dir, "/processed/malawi/dry_spells/dry_spells_during_rainy_season_list.csv"), row.names = FALSE)
+
+# summary stats per region 
```

Comment:
As I understand here region is referring to admin1? I would just be somewhat careful of the accuracy of results aggregated to a higher admin level than the original results are based on. As the results you have are sensitive to the shape of the admin2 boundaries, it's not a given that the results would be the same as if you had processed the chirps data and calculated the dry spells straight from the admin1 boundaries. 

Looks like just an exploration here, but maybe something to keep in mind.

Reply:
Good point. Yes I was computing at region (admin1) level simply to verify a WFP analysis that saw most dry spells in the South. We should keep it in mind if we want to do anything at adm1, thanks for raising.